cmake/SearchForStuff.cmake

```diff
@@ -675,6 +675,13 @@ else()
   set(HAVE_SPNAV FALSE)
 endif()
 
+########################################
+# Find python3, which is used by tools/check_test_ran.py
+find_package(PythonInterp 3)
```

Comment:
[PythonInterp is deprecated in CMake in version 3.12](https://cmake.org/cmake/help/latest/module/FindPythonInterp.html) (focal is 3.16 but other platforms could be broken with this if it gets removed at some point). It would be great to migrate it to other functions.

Reply:
I've added some logic to use `find_package(Python3)` for cmake 3.12+ in b772d92e28df6d4080fea0a10cbacfe77df342ae